### PR TITLE
Android

### PR DESCRIPTION
**Pull request by lucksus**
Description in README how to build for Android.
The wabt create wouldn't compile with the Android tool chains, so I just scoped it within the test since it's used only there.

### CODE REVIEW

File: core/src/nucleus/ribosome.rs [hunk 1]
```diff
@@ -95,6 +94,7 @@ pub fn call(wasm: Vec<u8>, function_name: &str) -> Result<Runtime, InterpreterEr
 
 #[cfg(test)]
 mod tests {
+    extern crate wabt;
```
Comment (sphinxc0re) [inline core/src/nucleus/ribosome.rs]: `extern crate` statement should only be at the very top of the root file. This is an established standard in the Rust community that make crate lookup for developers really easy

Comment (sphinxc0re) [reply core/src/nucleus/ribosome.rs]: @thedavidmeister Yeah, this makes this even more confusing. @lucksus If the crate is not used anywhere except the test environment, the compiler will take care of that. The compiler does LTO.

Comment (lucksus) [reply core/src/nucleus/ribosome.rs]: No, it's not about the compiler. When I switch the toolchain to armv7-linux-androideabi, wabt won't compile it's C bindings on install. That's why I had to do this.

Also, did you see that I've moved the crate definition up and out of this module? This convo has become superfluous ;)

**Comment by thedavidmeister** [comment]
@lucksus i put a followup at https://github.com/holochain/holochain-rust/issues/110 as i'm finding it a bit fiddly to get setup atm

**Comment by zippy** [comment]
Well, almost there, I updated the readme for linux, and almost got everything running except ran into a cmake problem:

```
$ cargo build --target armv7-linux-androideabi --release

   Compiling num-traits v0.2.5
   Compiling typenum v1.10.0
<snip>
   Compiling parity-wasm v0.27.6
   Compiling sha2 v0.7.1
error: failed to run custom build command for `wabt-sys v0.1.4`
process didn't exit successfully: `/home/eric/code/metacurrency/holochain/holochain-rust/target/release/build/wabt-sys-96cce9112ebbc680/build-script-build` (exit code: 101)
--- stdout
running: "cmake" "/home/eric/.cargo/registry/src/github.com-1ecc6299db9ec823/wabt-sys-0.1.4/wabt" "-DBUILD_TESTS=OFF" "-DCMAKE_INSTALL_PREFIX=/home/eric/code/metacurrency/holochain/holochain-rust/target/armv7-linux-androideabi/release/build/wabt-sys-127990043ea0e656/out" "-DCMAKE_C_FLAGS= -ffunction-sections -fdata-sections -fPIC -march=armv7-a -mthumb -mfpu=vfpv3-d16 -mfloat-abi=softfp" "-DCMAKE_C_COMPILER=arm-linux-androideabi-gcc" "-DCMAKE_CXX_FLAGS= -ffunction-sections -fdata-sections -fPIC -march=armv7-a -mthumb -mfpu=vfpv3-d16 -mfloat-abi=softfp" "-DCMAKE_CXX_COMPILER=arm-linux-androideabi-g++" "-DCMAKE_BUILD_TYPE=Release"
-- The C compiler identification is unknown
-- The CXX compiler identification is unknown
-- Configuring incomplete, errors occurred!
See also "/home/eric/code/metacurrency/holochain/holochain-rust/target/armv7-linux-androideabi/release/build/wabt-sys-127990043ea0e656/out/build/CMakeFiles/CMakeOutput.log".
See also "/home/eric/code/metacurrency/holochain/holochain-rust/target/armv7-linux-androideabi/release/build/wabt-sys-127990043ea0e656/out/build/CMakeFiles/CMakeError.log".

--- stderr
CMake Error at CMakeLists.txt:18 (project):
  The CMAKE_C_COMPILER:

    arm-linux-androideabi-gcc

  is not a full path and was not found in the PATH.

  Tell CMake where to find the compiler by setting either the environment
  variable "CC" or the CMake cache entry CMAKE_C_COMPILER to the full path to
  the compiler, or to the compiler name if it is in the PATH.


CMake Error at CMakeLists.txt:18 (project):
  The CMAKE_CXX_COMPILER:

    arm-linux-androideabi-g++

  is not a full path and was not found in the PATH.

  Tell CMake where to find the compiler by setting either the environment
  variable "CXX" or the CMake cache entry CMAKE_CXX_COMPILER to the full path
  to the compiler, or to the compiler name if it is in the PATH.


thread 'main' panicked at '
command did not execute successfully, got: exit code: 1

build script failed, must exit now', /home/eric/.cargo/registry/src/github.com-1ecc6299db9ec823/cmake-0.1.31/src/lib.rs:643:5
note: Run with `RUST_BACKTRACE=1` for a backtrace.

warning: build failed, waiting for other jobs to finish...
error: build failed
```

Clearly something needed to configure Cmake on linux, but not quite sure how to.

**Comment by thedavidmeister** [comment]
@lucksus FYI i noticed there's a brew cask for `android-ndk`, dunno if it's a valid alternative to the full studio approach:

`brew cask search android-ndk`

**Pull request merged**